src/index.js

```diff
@@ -1,47 +1,83 @@
 import './index.css';
 
-// Get DOM
-const listElement = document.createElement('div');
-listElement.classList = 'dynamic-container';
+let todoList = window.localStorage.getItem('todoList') ? JSON.parse(localStorage.getItem('todoList')) : [];
+const inputItem = document.querySelector('.inputs-field');
 const listContainer = document.querySelector('.list');
+const addButton = document.querySelector('.add-btn');
+const clearAll = document.querySelector('#clear-btn');
 
-const task = [
-  {
-    description: 'task1',
-    completed: false,
-    index: 1,
-  },
-  {
-    description: 'task2',
-    completed: false,
-    index: 2,
-  },
-  {
-    description: 'task3',
-    completed: false,
-    index: 3,
-  },
-  {
-    description: 'task4',
-    completed: false,
-    index: 4,
-  },
-];
-function iterateTasks() {
-  task.forEach((item) => {
-    listElement.innerHTML
-        += `
-        <div class="list-properties">
-            <span class="left-items">
-                <input type="checkbox">
-                <p>${item.description}</p>
-            </span>
-            <i class="fa-solid fa-ellipsis-vertical"></i>
-        </div>
-        <hr>
-        `;
+const displayToDoList = () => {
+  listContainer.innerHTML = '';
+  todoList.forEach((item, index) => {
+    const listItem = document.createElement('div');
+    listItem.className = 'list-properties';
+    listItem.innerHTML = `
+    <div class="both-sides">
+      <span class="left-items">
+        <input type="checkbox" ${item.completed ? 'checked' : ''}>
+        <p>${item.description}</p>
+      </span>
+      <span class="right-items">
+        <div id="remove-single"><i class="fa-solid fa-trash"></i></div>
+        <i class="fa-solid fa-ellipsis-vertical"></i>
+      </span>
+      </div>
+    `;
+    listItem.querySelector('#remove-single').addEventListener('click', (event) => {
+      event.preventDefault();
+      // eslint-disable-next-line no-use-before-define
+      removeItem(index);
+    });
+    listItem.querySelector('input[type="checkbox"]').addEventListener('change', (event) => {
+      event.preventDefault();
+      // eslint-disable-next-line no-use-before-define
+      markCompleted(index);
+    });
+    listContainer.appendChild(listItem);
   });
+};
 
-  listContainer.appendChild(listElement);
-}
-iterateTasks();
\ No newline at end of file
+const saveTodoList = () => {
+  localStorage.setItem('todoList', JSON.stringify(todoList));
+};
+const addItem = () => {
+  const task = inputItem.value;
+  if (task) {
+    const taskItem = {
+      description: task,
+      completed: false,
+    };
```

Comment:
- [ ] As suggested by the previous reviewer that all the tasks should have an index value 👍 to do that you can make your `taskItem` array something like the below 👇

```js
    const taskItem = {
      description: task,
      completed: false,
      index: todoList.length+1
    };
```